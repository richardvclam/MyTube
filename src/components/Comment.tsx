import React, { Component } from 'react';
import { Image, Text, View } from 'react-native';
import { AllHtmlEntities } from 'html-entities';

interface CommentProps {
  author: string;
  profileImageUrl: string;
  text: string;
}

export default class Comment extends Component<CommentProps> {
  render() {
    const { author, profileImageUrl, text } = this.props;

    const decodedText = new AllHtmlEntities().decode(text);

    return (
      <View style={{ padding: 15, flexDirection: 'row' }}>
        <Image
          source={{
            uri: profileImageUrl
          }}
          style={{ width: 45, height: 45, marginRight: 15, borderRadius: 23 }}
        />
        <View style={{ flex: 1 }}>
          <Text style={{ color: '#5e5e5e', fontSize: 14 }}>{author}</Text>
          <Text style={{ color: '#141414' }}>{decodedText}</Text>
        </View>
      </View>
    );
  }
}
