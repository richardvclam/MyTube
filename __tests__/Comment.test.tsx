import 'react-native';
import React from 'react';
import Comment from '../src/components/Comment';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly with an author, profile image, and text', () => {
  const tree = renderer
    .create(
      <Comment
        author="John"
        profileImageUrl="https://via.placeholder.com/150"
        text="Test comment"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
