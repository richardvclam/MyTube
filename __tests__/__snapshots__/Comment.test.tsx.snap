// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with an author, profile image, and text 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "padding": 15,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://via.placeholder.com/150",
      }
    }
    style={
      Object {
        "borderRadius": 23,
        "height": 45,
        "marginRight": 15,
        "width": 45,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#5e5e5e",
          "fontSize": 14,
        }
      }
    >
      John
    </Text>
    <Text
      style={
        Object {
          "color": "#141414",
        }
      }
    >
      Test comment
    </Text>
  </View>
</View>
`;
